master/buildbot/status/web/status_json.py

```diff
@@ -1371,6 +1382,25 @@ def asDict(self, request):
         defer.returnValue(build_number)
 
 
+class MyBuildsJsonResource(JsonResource):
+    help = 'Gives information about my builds from last 7 days'
+    pageTitle = 'My builds from last 7 days'
+
+    @defer.inlineCallbacks
+    def asDict(self, request):
+        authz = request.site.buildbot_service.authz
+        if 'user' in request.args and request.args['user'][0]:
+            username = request.args['user'][0]
+        else:
+            username = authz.getUsername(request)
+        user_info = authz.getUserInfo(username)
+        if user_info:
+            builds = yield MybuildsResource.prepare_builds(self.status.master, user_info['uid'])
+            defer.returnValue(builds)
+        else:
+            defer.returnValue(['User "%s" is unknown' % username])
```

Comment:
It would be nice to change it for dictionary with 'error' key.

Reply:
Done